code/game/machinery/ringer.dm

```diff
@@ -0,0 +1,121 @@
+/obj/machinery/ringer
+	name = "ringer terminal"
+	desc = "A ringer terminal, PDAs can be linked to it."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "bell_standby"
+	anchored = 1.0
+
+	req_access = list(access_hop) //what access it needs to link your pda
+
+	var/id = null
+	var/list/obj/item/device/pda/rings_pdas = list() //A list of PDAs to alert upon someone touching the machine
+	var/_wifi_id
+	var/datum/wifi/receiver/button/ringer/wifi_receiver
+	var/on = TRUE
+	var/department = "Hop's harem" //whatever department/desk you put this thing
+
+/obj/machinery/ringer/Initialize()
+	. = ..()
+	if(_wifi_id)
+		wifi_receiver = new(_wifi_id, src)
+
+/obj/machinery/ringer/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/ringer/Destroy()
+	qdel(wifi_receiver)
+	wifi_receiver = null
+	return ..()
+
+/obj/machinery/ringer/update_icon()
+	if(stat & NOPOWER)
+		icon_state = "bell_off"
+		return
+	if(!on)
+		icon_state = "bell_off"
+	if (rings_pdas || rings_pdas.len)
+		icon_state = "bell_active"
+	else
+		icon_state = "bell_standby"
+
+/obj/machinery/ringer/attackby(obj/item/C as obj, mob/living/user as mob)
+	if(stat & (BROKEN|NOPOWER) || !istype(usr,/mob/living))
+		return
+
+	if (istype(C, /obj/item/device/pda))
+		if(!check_access(C))
+			user << "<span class='warning'>Access Denied.</span>"
+			return
+		else if (C in rings_pdas)
+			usr << "<span class='notice'>\The [C] appears to be already linked.</span>"
+			return
+		usr << "<span class='notice'>You link \the [C] to \the [src], it will now ring upon someone using \the [src].</span>"
+		rings_pdas += C
```

Comment:
This should probably be a weakref or use OnDestroy to remove the PDA on destruction.